modules/widgets/simple-payments.php

```diff
@@ -266,6 +266,21 @@ public function ajax_delete_payment_button() {
 			wp_send_json_success( $return );
 		}
 
+		/**
+		 * Returns the number of decimal places on string representing a price.
+		 *
+		 * @param string $number Price to check.
+		 * @return number number of decimal places.
+		 */
+		private function get_decimal_places( $number ) {
+			preg_match( '/(?:\.(\d+))/', $number, $match );
+			if ( count( $match ) <= 1 ) {
+				return 0;
+			}
+
+			return isset( $match[1] ) ? strlen( $match[1] ) : 0;
```

Comment:
note that we can separate the question "did we find matches" from "what are the matches"
further, in cases where we end up referencing array values by index I find it helpful to give names to those positions

```php
$match = preg_match( '/(?:\.(\d+))/', $number, $matches );

// no digits found after a decimal could plausibly also be `false`
// but that would mean we have a bad pattern here which we don't
if ( 0 === $match ) {
	return 0;
}

list( /* full match */, $digits ) = $matches;
return strlen( $digits );
```

we may want to clarify the input domain to make the pattern less ambiguous or even constrain the pattern more.
are we expecting a string representing a currency value _with_ a currency sign, like `$10.13` or `0,50€` or `100¥`?

are we expecting a number string that _should_ be convertible into a floating-point number?

```php
preg_match( '/^\d+\.(\d+)$/', … )
```

more importantly, not all currencies use the decimal point as a fractional separator. it's common to see something like this: `1.750.000,49€` If we are accepting localized strings then we need to account for that